libs/@blockprotocol/graph/src/codegen/parameters.ts

```diff
@@ -0,0 +1,241 @@
+          ) {
+            errors.push(
+              `if an entry under file '${fileName}' in 'targets' has a 'blockEntity' flag, it must be a boolean`,
+            );
+          }
+        }
+      }
+    }
+  }
+
+  if (typeNameOverrides !== undefined) {
+    if (
+      !(typeof typeNameOverrides === "object" && typeNameOverrides !== null)
+    ) {
+      errors.push("`typeNameOverrides` must be an object");
+    } else {
+      for (const [versionedUrl, typeNameOverride] of Object.entries(
+        typeNameOverrides,
+      )) {
+        if (!validateVersionedUrl(versionedUrl)) {
+          errors.push(
+            `each key in 'typeNameOverrides' must be a valid Versioned URL`,
+          );
+        }
+        if (typeof typeNameOverride !== "string") {
+          errors.push(
+            `each entry in 'typeNameOverrides' must be a string, but '${versionedUrl}' is a ${typeof typeNameOverride}`,
+          );
+        }
+      }
+    }
+  }
+
+  if (versionedUrlAliases !== undefined) {
+    if (
+      !(typeof versionedUrlAliases === "object" && versionedUrlAliases !== null)
+    ) {
+      errors.push("`versionedUrlAliases` must be an object");
+    } else if (
+      !("enabled" in versionedUrlAliases) ||
+      versionedUrlAliases.enabled === undefined
+    ) {
+      errors.push("`versionedUrlAliases.enabled` must be set");
+    } else if (typeof versionedUrlAliases.enabled !== "boolean") {
+      errors.push("`versionedUrlAliases.enabled` must be a boolean");
+    } else if (versionedUrlAliases.enabled) {
+      if (
+        "overrides" in versionedUrlAliases &&
+        versionedUrlAliases.overrides !== undefined
+      ) {
+        if (
+          !(
+            typeof versionedUrlAliases.overrides === "object" &&
+            versionedUrlAliases.overrides !== null
+          )
+        ) {
+          errors.push("`versionedUrlAliases.overrides` must be an object");
+        } else {
+          for (const [versionedUrl, alias] of Object.entries(
+            versionedUrlAliases.overrides,
+          )) {
+            if (!validateVersionedUrl(versionedUrl)) {
+              errors.push(
+                `each key in 'versionedUrlAliases.overrides' must be a valid Versioned URL`,
+              );
+            }
+            if (typeof alias !== "string") {
+              errors.push(
+                `each entry in 'versionedUrlAliases.overrides' must be a string, but '${versionedUrl}' is a ${typeof alias}`,
+              );
+            }
+          }
+        }
+      }
+    }
+  }
+
+  if (temporal !== undefined && typeof temporal !== "boolean") {
+    errors.push("`temporal` must be a boolean");
+  }
+
+  if (errors.length > 0) {
+    return { errors };
+  }
+};
+
+export type ProcessedCodegenParameters = Omit<
+  Required<CodegenParameters>,
+  "targets"
+> & {
+  targets: {
+    [fileName: string]: {
+      versionedUrls: VersionedUrl[];
+      blockEntity?: VersionedUrl;
+    };
```

Comment:
I opted for a different representation here because it's easier to deal with later on, but I wanted to avoid making users copy paste the same Versioned URL twice in the package.json, as opposed to just being able to write an inline flag like `blockEntity: true`